internal/service/project/resource_project.go

```diff
@@ -549,22 +541,22 @@ func getProjectPropsFromAPI(ctx context.Context, conn *matlas.Client, connV2 *ad
 		return nil, nil, nil, fmt.Errorf("error getting project's limits (%s): %s", projectID, err.Error())
 	}
 
-	projectSettings, _, err := conn.Projects.GetProjectSettings(ctx, projectID)
+	projectSettings, _, err := connV2.ProjectsApi.GetProjectSettings(ctx, projectID).Execute()
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("error getting project's settings assigned (%s): %v", projectID, err.Error())
 	}
 
 	return teams, limits, projectSettings, nil
 }
 
-func newTFProjectResourceModel(ctx context.Context, projectRes *matlas.Project,
-	teams *matlas.TeamsAssigned, projectSettings *matlas.ProjectSettings, limits []admin.DataFederationLimit) *tfProjectRSModel {
+func newTFProjectResourceModel(ctx context.Context, projectRes *admin.Group,
+	teams *admin.PaginatedTeamRole, projectSettings *admin.GroupSettings, limits []admin.DataFederationLimit) *tfProjectRSModel {
 	projectPlan := tfProjectRSModel{
-		ID:                        types.StringValue(projectRes.ID),
+		ID:                        types.StringValue(*projectRes.Id),
```

Comment:
don't we need to use an auxiliar function (o calling getId) in case pointer is nil?

Reply:
Changed it to use the getter